independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/Instances.java

```diff
@@ -32,12 +34,16 @@ public static List<InjectableBean<?>> resolveBeans(Type requiredType, Set<Annota
     }
 
     public static List<InjectableBean<?>> resolveBeans(Type requiredType, Annotation... requiredQualifiers) {
-        return ArcContainerImpl.instance()
-                .getResolvedBeans(requiredType, requiredQualifiers)
-                .stream()
-                .filter(Predicate.not(InjectableBean::isSuppressed))
-                .sorted(PRIORITY_COMPARATOR)
-                .collect(Collectors.toUnmodifiableList());
+        Set<InjectableBean<?>> resolvedBeans = ArcContainerImpl.instance()
+                .getResolvedBeans(requiredType, requiredQualifiers);
+        List<InjectableBean<?>> nonSuppressed = new ArrayList<>(resolvedBeans.size());
+        for (InjectableBean<?> injectableBean : resolvedBeans) {
+            if (!injectableBean.isSuppressed()) {
+                nonSuppressed.add(injectableBean);
+            }
+        }
+        nonSuppressed.sort(PRIORITY_COMPARATOR);
+        return Collections.unmodifiableList(nonSuppressed);
```

Comment:
Hm, this is not equivalent in the sense that previously an optimized immutable list (`List.of()`) was used whereas here we use an unmodifiable wrapper...

Reply:
It's equivalent for all intents and purposes, no?

Reply:
It's not equivalent in terms of memory consumption because in theory there will be a lot resolved lists of beans that are stored in the cache and `Collections.unmodifiableList(new ArrayList<>())` is definitely worse compared to `List.of()`.

Reply:
But that `List.of` is produced by a `Stream` which itself takes up memory to be processed.

Reply:
> But that `List.of` is produced by a `Stream` which itself takes up memory to be processed.

Which is freed afterwards but the list is there to stay as a cached value.

I was thinking about replacing the `Collections.unmodifiableList(nonSuppressed)` with `List.copyOf(nonSuppressed)`...

Reply:
No problem, done